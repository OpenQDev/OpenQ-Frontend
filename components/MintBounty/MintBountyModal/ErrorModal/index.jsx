// Third party
import React from 'react';
import ModalDefault from '../../../Utils/ModalDefault';

const ErrorModal = ({ setShowErrorModal, error }) => {
  const { title, message } = error;

  const updateModal = () => {
    setShowErrorModal();
  };

  const btn = (
    <button className='btn-default w-full' type='button' onClick={() => updateModal()}>
      Close
    </button>
  );

  return (
    <ModalDefault title={title} footerRight={btn} setShowModal={setShowErrorModal} resetState={updateModal}>
      <div>{message}</div>
    </ModalDefault>
  );
};

export default ErrorModal;
