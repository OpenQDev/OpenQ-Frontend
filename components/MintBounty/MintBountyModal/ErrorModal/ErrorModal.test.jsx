import React from 'react';
import ErrorModal from '.';
import renderer from 'react-test-renderer';

describe('ErrorModal', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<ErrorModal error={{ title: 'Error', message: 'My error message' }} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
