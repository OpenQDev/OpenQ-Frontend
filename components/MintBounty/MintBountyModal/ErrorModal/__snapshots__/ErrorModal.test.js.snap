// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal should match DOM Snapshot 1`] = `
<div>
  <div
    className="flex justify-center items-end sm:items-center overflow-x-hidden overflow-y-auto fixed inset-0 z-[53]"
  >
    <div
      className="flex w-full sm:w-[480px] h-[320px] border border-gray-700 sm:rounded-sm bg-[#161B22]"
    >
      <div
        className="flex flex-col relative w-full"
      >
        <button
          className="absolute top-4 right-4 cursor-pointer"
          data-testid="cross"
          onClick={[Function]}
        >
          <svg
            className="fill-muted hover:fill-blue-400 w-4 h-4"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
              fillRule="evenodd"
            />
          </svg>
        </button>
        <div
          className="py-2 px-4 border-b border-gray-700 text-2xl w-full h-[50px]"
        >
          Error
        </div>
        <div
          className="py-2 px-4 w-full h-[220px] overflow-x-hidden overflow-y-auto"
        >
          <div>
            My error message
          </div>
        </div>
        <div
          className="flex justify-end  py-2 px-4 border-t border-gray-700 w-full h-[50px]"
        >
          <div />
          <div>
            <button
              className="btn-default w-full"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bg-overlay z-10 fixed inset-0"
  />
</div>
`;
